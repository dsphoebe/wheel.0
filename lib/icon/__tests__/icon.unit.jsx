import * as renderer from 'react-test-renderer';
import React from 'react';
import { mount } from 'enzyme';
import Icon from '../icon';

describe('icon', () => {
    it('render Icon', () => {
        const json = renderer.create(<Icon name={"wechat"} />).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick', () => {
        const fn = jest.fn()
        const c = mount(<Icon name={"wechat"} onClick={ fn }/>)
        c.find('svg').simulate('click')
        expect(fn).toBeCalled();
    })
})
