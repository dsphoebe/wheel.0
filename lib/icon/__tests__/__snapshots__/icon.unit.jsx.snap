// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render Icon 1`] = `
<svg
  className="wui-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
